source/MaterialXGenMdl/MdlSyntax.cpp

```diff
@@ -544,4 +545,42 @@ string MdlSyntax::getArrayTypeSuffix(const TypeDesc* type, const Value& value) c
     return string();
 }
 
+bool MdlSyntax::remapEnumeration(const string& value, const TypeDesc* type, const string& enumNames, std::pair<const TypeDesc*, ValuePtr>& result) const
+{
+    // Early out if not an enum input.
+    if (enumNames.empty())
+    {
+        return false;
+    }
+
+    // Don't convert or filenames and arrays.
+    if (type == Type::FILENAME || (type && type->isArray()))
```

Comment:
Nitpick: Is the nullptr check on `type` needed, it was not done in the GlslSyntax version.

Reply:
I will fix GlslSyntax since it should check null.